### PR TITLE
If `at` command is not present on the target host attempt to use `nohup`

### PR DESCRIPTION
**Pull request by konstan**
If `at` command is not present on the target host attempt to use `nohup` (if present), or default to just going to background.  In the latter case it's better if the target script(/command) traps `HUP` signal in the case the ssh connection was requesting controlling terminal.  In this case, the script will receive the signal from the shell that is going to be terminated by the logout, and effectively ignores it.

Connected to #193

### CODE REVIEW

File: client/src/main/python/slipstream/utils/ssh.py [hunk 1]
```diff
@@ -394,15 +394,31 @@ def remoteRunScript(user, host, script, sshKey=None, password='', nohup=False):
...
+                raise Exceptions.ExecutionException("An error occurred while executing the command: %s\n%s." %
+                                                    (command, stderr))
+        else:
+            raise Exceptions.ExecutionException("An error occurred while executing the command: %s\n%s." %
+                                                (command, stderr))
+    else:
+        # Check stderr as atd may not be running, though return code was 0.
+        # Starting atd service will start the command.
+        if re.search('.*No atd running\?', stderr, re.MULTILINE):
+            remoteRunCommand(user, host, 'service atd start', sshKey, password)
```
Comment (schaubl) [inline client/src/main/python/slipstream/utils/ssh.py]: This line may not work on all Linux distribution.

Comment (konstan) [inline client/src/main/python/slipstream/utils/ssh.py]: Yes, agreed this is a tricky bit.  I'll see tomorrow.

**Comment by konstan** [comment]
According to https://www.digitalocean.com/community/tutorials/how-to-configure-a-linux-service-to-start-automatically-after-a-crash-or-reboot-part-1-practical-examples

- `System V`
  - Debian 6 and earlier
  - Ubuntu 9.04 and earlier
  - CentOS 5 and earlier
- `Upstart`
  - Ubuntu 9.10 to Ubuntu 14.10, including Ubuntu 14.04
  - CentOS 6
- `systemd`
  - Debian 7 and Debian 8
  - Ubuntu 15.04
  - CentOS 7

`System V` - `service <service> start`
`Upstart` - `initctl start <service>`
`systemd` - `systemctl start <service>`

Highly likely we can assume that on current CentOS 6.x and Ubuntu 14.x we can still use `System V`.  So, we are left with supporting `System V` and `systemd`.  If you @schaubl are agree to that I'll proceed with an implementation.

**Comment by schaubl** [comment]
BTW: I know that on Ubuntu you can use *Upstart* with `service <service> ...`

Why not simply do something like this:
```bash
remoteRunCommand(user, host, 'service atd start || initctl start atd || systemctl start atd || /etc/init.d/atd start', sshKey, password)
```
?

**Comment by konstan** [comment]
It even works on CentOS 7 with `systemd` (at least for now via "redirect").

```
[root@vm ~]# cat /etc/centos-release
CentOS Linux release 7.0.1406 (Core)
[root@vm ~]# service atd status
Redirecting to /bin/systemctl status  atd.service
atd.service - Job spooling tools
   Loaded: loaded (/usr/lib/systemd/system/atd.service; enabled)
   Active: inactive (dead)

[root@vm ~]# service atd start
Redirecting to /bin/systemctl start  atd.service
[root@vm ~]# service atd status
Redirecting to /bin/systemctl status  atd.service
atd.service - Job spooling tools
   Loaded: loaded (/usr/lib/systemd/system/atd.service; enabled)
   Active: active (running) since Tue 2015-11-17 10:47:31 UTC; 3s ago
 Main PID: 11438 (atd)
   CGroup: /system.slice/atd.service
           └─11438 /usr/sbin/atd -f

Nov 17 10:47:31 vm.nohup-ssh systemd[1]: Started Job spooling tools.
[root@vm ~]#
```
So, it looks as if even with `service <service> start` we are safe at least on the mainstream distros. 

Anyways, I'll augment the command with what you are suggesting.  Thank you.

**Pull request merged**